cmd/meroxa/turbine/utils.go

```diff
@@ -182,18 +184,43 @@ func GitInit(ctx context.Context, appPath string) error {
 		return errors.New("path is required")
 	}
 
-	cmd := exec.Command("git", "config", "--global", "init.defaultBranch", "main")
-	cmd.Path = appPath
-	_ = cmd.Run()
+	oldWay := checkGitVersion(ctx)
+
+	if !oldWay {
+		cmd := exec.CommandContext(ctx, "git", "config", "--global", "init.defaultBranch", "main")
+		cmd.Path = appPath
+		_ = cmd.Run()
+	}
 
-	cmd = exec.Command("git", "init", appPath)
+	cmd := exec.CommandContext(ctx, "git", "init", appPath)
 	output, err := cmd.CombinedOutput()
 	if err != nil {
 		return fmt.Errorf(string(output))
 	}
+
+	if oldWay {
+		cmd := exec.CommandContext(ctx, "git", "checkout", "-b", "main")
+		cmd.Path = appPath
+		_ = cmd.Run()
+	}
 	return nil
 }
 
+func checkGitVersion(ctx context.Context) bool {
```

Comment:
I think we should return error on `cmd.CombinedOutput`. Aside, I wonder we could make this a bit clearer by making these tweaks.

@janelletavares what do you think?

```suggestion
func checkGitVersion(ctx context.Context) (bool, error) {
	cmd := exec.CommandContext(ctx, "git", "version")
	out, err := cmd.CombinedOutput()
	if err != nil {
	  return false, err
	}

	r := regexp.MustCompile("git version ([0-9.]+)")
	matches := r.FindStringSubmatch(string(out))
	if len(matches) > 0 {
		comparison := semver.Compare("2.28", matches[1])
		return comparison >= 1
	}
	return true
}
```

Reply:
happy to change the returns

i mentioned above that i wanted to error here as little as possible and just have some kind of sane default behavior.  i could go either way.